docs/practices/start.md

```diff
@@ -0,0 +1,127 @@
+# How to start developing nearcore
+
+## Running a local node
+
+If you're looking to run a validator or RPC node in mainnet or testnet, you can follow this [article](https://near-nodes.io/validator/running-a-node) that is using nearup or [this article](https://docs.near.org/pt-BR/docs/develop/node/validator/compile-and-run-a-node/) that shows you how to compile and run it.
+
+In this section, we'll cover the preferred setup that you might want to have when you do the local nearcore development.
+
+### Step 1 - get the code
+
+```shell
+git clone https://github.com/near/nearcore
+
+git checkout master
+```
+
+### Step 2 - compilation
+
+There are 3 options:
+
+* default (debug) binary - faster compilation, slower running
+```shell
+cargo build --package neard
+```
+
+* quick-release - medium compilation speed, faster running
+```shell
+cargo build --package neard --profile quick-release
+```
+
+* release - slow compilation time, even faster running
+```shell
+cargo build --package neard --release
+```
+
+IMPORTANT: each one of the methods above, is putting a binary in a different location (``target/debug``, ``target/quick-release``, ``target/release``) - so make sure that you adjust other scripts accordingly.
+
+For local development, we'd recomment using the default (debug) build.
+
+
+### Step 3 - starting the node
+
+There are couple ways to start the locanet, but the way I'd recommend is to use the nearup. This way you'll be able to automatically start 4 nodes on your local machine, therefore also testing the networking code, synchronization etc.
+
+First: get the nearup:
+
+```
+git clone https://github.com/near/nearup
+cd nearup
+```
+
+And then you can start the node using the command below (from the nearup directory):
+
+```
+python3 nearup run localnet --binary-path ../nearcore/target/debug/ --interactive
+```
+
+The command above, is going to ask you a bunch of questions.
+
+* How many nodes -- the default is 4 - and I'd recommend that if your machine can handle it.
+* How many shards - the default is 1 - but I'd recomment actually running with 4 shards.
+* Would you like a fixed accounts - I'd say yes - this way you'll automatically have a bunch of accounts that are spread between different shards (for example shard0 will be on shard 0, shard1 will be on shard1 etc).
+* Should nodes be archival - up to you (archival nodes will take some more disk space - but in most localnet cases, thats negligible difference).
+
+Then you can go to http://localhost:3030/debug and congratulations - you are officially running a localnet instance.
+
+If you want to shut it down, you can run:
+
+```shell
+python3 nearup stop
+```
```

Comment:
nearup is deprecated as far as i understand: 

https://github.com/near/nearcore/issues/7876

Reply:
And, from contribution easiness perspective, we should avoid python, installing python deps is a huge stumbling block.

Reply:
And from the educational prespective, it would help if you manually start all four nodes, otherwise nearup would look like magic.